fossunited/foss_profiles/doctype/foss_user_profile/test_foss_user_profile.py

```diff
@@ -36,3 +39,36 @@ def test_add_profile(self):
         profile_exists = frappe.db.exists(USER_PROFILE, {"user": frappe_user.name})
 
         self.assertTrue(profile_exists)
+
+    def test_private_profile_access(self):
+        fake = Faker()
+        test_name = fake.name()
+        test_user = frappe.get_doc(
+            {
+                "doctype": "User",
+                "email": str(uuid.uuid4()) + "@fossunited.org",
+                "first_name": test_name.split(" ")[0],
+                "name": test_name,
```

Comment:
nit: the "name" is reserved for "IDs" in frappe. For User doctype, frappe auto-sets the name to be the email of the user. 

So even this is not set, frappe will set the name to be the email


Alt solution:

```py
test_email = fake.email()
test_user = frappe.get_doc(
   {
       "doctype": "User",
       "email": test_email,
       "first_name": fake.name().split(" ")[0],
    }
).insert()

private_profile = frappe.get_doc(USER_PROFILE, {"user": test_email})
```

Reply:
@harshtandiya 
```py
private_profile = frappe.get_doc(USER_PROFILE, {"user": test_email})
```
Shouldn't we add `"is_private": 1` too?

Reply:
if it's a nitpicky comment, i'd prefer if we fix it ourselves later, instead of delaying the external contribution further.

thanks for the clarification!